lib/ui/platform_dispatcher.dart

```diff
@@ -0,0 +1,856 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.6
+part of dart.ui;
+
+typedef PlatformConfigurationChangedCallback = void Function(PlatformConfiguration/*!*/ configuration);
+
+/// Platform event dispatcher singleton.
+///
+/// The most basic interface to the host operating system's interface.
+///
+/// This is the central entry point for platform messages and configuration
+/// events from the platform.
+///
+/// It exposes the he core scheduler API, the input event callback, the graphics
+/// drawing API, and other such core services.
+///
+/// It manages the list of the application's [views] and the [screens] attached
+/// to the device, as well as the [configuration] of various platform
+/// attributes.
+///
+/// Please try to avoid statically referencing this singleton though
+/// [PlatformDispatcher.instance] and instead use a binding for dependency
+/// resolution such as `WidgetsBinding.instance.platformDispatcher`. See
+/// [PlatformDispatcher.instance] for more information about why this is
+/// preferred.
+class PlatformDispatcher {
+  /// Private constructor, since only dart:ui is supposed to create one of
+  /// these.
+  PlatformDispatcher._() {
+    _setNeedsReportTimings = _nativeSetNeedsReportTimings;
+  }
+
+  /// The [PlatformDispatcher] singleton.
+  ///
+  /// Please try to avoid statically referencing this and instead use a binding
+  /// for dependency resolution such as
+  /// `WidgetsBinding.instance.platformDispatcher`.
+  ///
+  /// Static access of this object means that Flutter has few, if any options to
+  /// fake or mock the given object in tests. Even in cases where Dart offers
+  /// special language constructs to forcefully shadow such properties, those
+  /// mechanisms would only be reasonable for tests and they would not be
+  /// reasonable for a future of Flutter where we legitimately want to select an
+  /// appropriate implementation at runtime.
+  ///
+  /// The only place that `WidgetsBinding.instance.platformDispatcher` is
+  /// inappropriate is if access to these APIs is required before invoking
+  /// `runApp()`. In that case, it is acceptable (though unfortunate) to use the
+  /// [PlatformDispatcher.instance] object statically.
```

Comment:
no if you want to call these before runApp you just instantiate the binding yourself.

The only time it's appropriate to access these directly rather than through the binding is if you're implementing a binding.